[GRPC] Documentation for 3.2 GRPC GA (Bulk and k-NN Search) (#10651)
opensearch-project/documentation-website: _api-reference/grpc-apis/search.md

| `ext` | [`ObjectMap`](https://github.com/opensearch-project/opensearch-protobufs/blob/0.6.0/protos/schemas/common.proto#L76)                      | Plugin-specific metadata, for example, for extensions like RAG. |
| `from` | `int32`                                                                                                                                    | The starting index for paginated results. Default is `0`. |
| `highlight` | [`Highlight`](https://github.com/opensearch-project/opensearch-protobufs/blob/0.6.0/protos/schemas/common.proto#L516)                     | Highlights matched terms in the result snippets. |
| `track_total_hits` | [`TrackHits`](https://github.com/opensearch-project/opensearch-protobufs/blob/0.6.0/protos/schemas/search.proto#L309)                     | Whether to return the total hit count. |
| `indices_boost` | `repeated` [`NumberMap`](https://github.com/opensearch-project/opensearch-protobufs/blob/0.6.0/protos/schemas/search.proto#L747)          | Per-index boost multipliers in the `<index>: <boost>` format. |

> | `docvalue_fields` | `repeated` [`FieldAndFormat`](https://github.com/opensearch-project/opensearch-protobufs/blob/0.6.0/protos/schemas/common.proto#L893)     | The fields returned using doc values, optionally formatted. |

Reviewer: ```suggestion
| `docvalue_fields` | `repeated` [`FieldAndFormat`](https://github.com/opensearch-project/opensearch-protobufs/blob/0.6.0/protos/schemas/common.proto#L893)     | The fields returned using doc values. Optionally, this field can be formatted for readability. |
```

### DerivedField fields

The [`DerivedField`](https://github.com/opensearch-project/opensearch-protobufs/blob/0.6.0/protos/schemas/search.proto#L292) message is used for dynamically computed fields that are calculated during search execution. It accepts the following fields.

| Field | Protobuf type | Description |
| :---- | :---- | :---- |
| `name` | `string` | The name of the derived field. Required. |
| `type` | `string` | The data type of the derived field. Required. |
| `script` | [`Script`](https://github.com/opensearch-project/opensearch-protobufs/blob/0.6.0/protos/schemas/common.proto#L27) | The script that computes the field value. Required. |
| `prefilter_field` | `string` | The field to use for prefiltering to optimize script execution. Optional. |
| `properties` | `map<string, `[`ObjectMap`](https://github.com/opensearch-project/opensearch-protobufs/blob/0.6.0/protos/schemas/common.proto#L76)`>` | Additional properties for the derived field. Optional. |
| `ignore_malformed` | `bool` | Whether to ignore malformed values during field computation. Optional. |

> | `format` | `string` | The format to apply to the field value (e.g., date formatting). Optional. |

Reviewer: ```suggestion
| `format` | `string` | The format to apply to the field value (for example., date formatting). Optional. |
```

### DerivedField fields

The [`DerivedField`](https://github.com/opensearch-project/opensearch-protobufs/blob/0.6.0/protos/schemas/search.proto#L292) message is used for dynamically computed fields that are calculated during search execution. It accepts the following fields.

| Field | Protobuf type | Description |
| :---- | :---- | :---- |
| `name` | `string` | The name of the derived field. Required. |
| `type` | `string` | The data type of the derived field. Required. |
| `script` | [`Script`](https://github.com/opensearch-project/opensearch-protobufs/blob/0.6.0/protos/schemas/common.proto#L27) | The script that computes the field value. Required. |
| `prefilter_field` | `string` | The field to use for prefiltering to optimize script execution. Optional. |
| `properties` | `map<string, `[`ObjectMap`](https://github.com/opensearch-project/opensearch-protobufs/blob/0.6.0/protos/schemas/common.proto#L76)`>` | Additional properties for the derived field. Optional. |
| `ignore_malformed` | `bool` | Whether to ignore malformed values during field computation. Optional. |

> | `format` | `string` | The format to apply to the field value (for example., date formatting). Optional. |

Reviewer 2: ```suggestion
| `format` | `string` | The format to apply to the field value (for example, date formatting). Optional. |
```
